src/app/search/state/search.service.ts

```diff
@@ -271,10 +304,17 @@ export class SearchService {
 
   sortByAlphabet(orderedArray, orderMode): any {
     orderedArray = orderedArray.sort((a, b) => {
+      if (!a.key) {
+        return 1;
+      } // always put null/empty values at the end regardless of sort direction
+      if (!b.key) {
+        return -1;
+      }
+
       if (orderMode) {
-        return a.key > b.key ? 1 : -1;
+        return a.key.toLowerCase() > b.key.toLowerCase() ? 1 : -1; // ascending
```

Comment:
Why did you change this one? Won't this change the behavior of existing sorts?

Not saying you shouldn't have just trying to understand -- as I see it your compareAttributes is handling the case for author and name. What is the case for?

Reply:
I thought we would want to ignore case in other sorts, but you're right this isn't needed for this PR. No problem to undo it